lib/osml/model_runner/roles/mr_task_role.ts

```diff
@@ -74,72 +85,206 @@ export class MRTaskRole extends Construct {
+      resources: ["*"]
+    });
+
+    // Add permissions for SQS permissions
+    const sqsPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        "sqs:DeleteMessage",
+        "sqs:ListQueues",
+        "sqs:GetQueueUrl",
+        "sqs:ReceiveMessage",
+        "sqs:SendMessage",
+        "sqs:GetQueueAttributes"
+      ],
+      resources: [
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_IMAGE_REQUEST_QUEUE}`,
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_REGION_REQUEST_QUEUE}`,
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_IMAGE_STATUS_QUEUE}`,
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_REGION_STATUS_QUEUE}`,
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_IMAGE_REQUEST_QUEUE}DLQ`,
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_REGION_REQUEST_QUEUE}DLQ`,
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_IMAGE_STATUS_QUEUE}DLQ`,
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_REGION_STATUS_QUEUE}DLQ`
+      ]
+    });
+
+    // Add permissions for S3 permissions
+    const s3PolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        "s3:GetBucketAcl",
+        "s3:ListBucket",
+        "s3:GetBucketLocation",
+        "s3:GetObject",
+        "s3:GetObjectAcl",
+        "s3:PutObject",
+        "s3:DeleteObject"
+      ],
+      resources: [`arn:${this.partition}:s3:::*`]
+    });
+
+    // Add permissions for SNS permissions
+    const snsPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["sns:Publish"],
+      resources: [
+        `arn:${this.partition}:sns:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SNS_IMAGE_STATUS_TOPIC}`,
+        `arn:${this.partition}:sns:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SNS_REGION_STATUS_TOPIC}`
+      ]
+    });
+
+    // Add permissions for dynamodb permissions
+    const ddbPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        "dynamodb:BatchGetItem",
+        "dynamodb:BatchWriteItem",
+        "dynamodb:PutItem",
+        "dynamodb:ListTables",
+        "dynamodb:DeleteItem",
+        "dynamodb:GetItem",
+        "dynamodb:Query",
+        "dynamodb:Scan",
+        "dynamodb:UpdateItem",
+        "dynamodb:UpdateTable"
+      ],
+      resources: [
+        `arn:${this.partition}:dynamodb:${props.account.region}:${props.account.id}:table/${this.mrDataplaneConfig.DDB_JOB_STATUS_TABLE}`,
```

Comment:
You could define local variables to make this a bit cleaner above each policy... like ''const status_table = this.mrDataplaneConfig.DDB_JOB_STATUS_TABLE''

Reply:
Fixed in next commit.